dev/fattest.simplicity/src/componenttest/topology/utils/DDLGenScript.java

```diff
@@ -0,0 +1,313 @@
+            }
+
+            String userDir = script.getServer().getUserDir();
+            String serverName = script.getServer().getServerName();
+
+            String _ddlDir;
+            if (userDir.length() >= 1 && userDir.endsWith(File.separator)) {
+                _ddlDir = userDir + "servers" + File.separator + serverName + File.separator + "ddl";
+            } else {
+                _ddlDir = userDir + File.separator + "servers" + File.separator + serverName + File.separator + "ddl";
+            }
+
+            ddlDir = new File(_ddlDir);
+
+        }
+
+        ///// Assertions /////
+
+        /**
+         * Asserts that the script(s) all ran successfully, that the expected success message was output,
+         * and that the directory where the DDL files should have been placed exists.
+         *
+         * @return this
+         */
+        public DDLGenScriptResult assertSuccessful() {
+            Deque<ProgramOutput> outputs = Objects.requireNonNull(script.getOutputs());
+
+            assertEquals("DDLGen script failed with result: " + RESULT_CODE_MAP.get(outputs.getFirst().getReturnCode()), 0, outputs.getFirst().getReturnCode());
+
+            Iterator<ProgramOutput> i = outputs.iterator();
+            while (i.hasNext()) {
+                ProgramOutput output = i.next();
+                assertEquals("Debug script failed with result: " + output.getReturnCode(), 0, output.getReturnCode());
+            }
+
+            sysout.stream()
+                            .filter(line -> line.contains(SUCCESS_MESSAGE))
+                            .findFirst()
+                            .orElseThrow(() -> new AssertionError("Output did not contain " + SUCCESS_MESSAGE));
+
+            assertTrue("The output directory did not exist: " + ddlDir.getAbsolutePath(), ddlDir.exists());
+            assertTrue("The output directory was not a directory: " + ddlDir.getAbsolutePath(), ddlDir.isDirectory());
+
+            return this;
+        }
+
+        /**
+         * Asserts that an expected DDL file exists in the DDL directory
+         *
+         * @param  expectedFileName the expected file name with .ddl extension
+         * @return                  this
+         */
+        public DDLGenScriptResult assertDDLFile(String expectedFileName) {
+            assertTrue("Expected file " + expectedFileName + " did not exist in " + ddlDir.getAbsolutePath(), getFileNames().contains(expectedFileName));
+            return this;
+        }
+
+        /**
+         * Asserts that all the expected DDL files exists in the DDL directory
+         *
+         * @param  expectedFileNames the list of expected file names with .ddl extensions
+         * @return                   this
+         */
+        public DDLGenScriptResult assertDDLFiles(List<String> expectedFileNames) {
+            assertEquals("Expected number of ddl files did not match actual number of files", expectedFileNames.size(), getFileNames());
```

Comment:
This was comparing a number of files to a list of files. In the suggestion to correct that, I also added the list of observed files to the error message to help with debugging.
```suggestion
            List<String> observedFileNames = getFileNames();
            assertEquals("Expected number of ddl files did not match actual number of files " + observedFileNames,
                         expectedFileNames.size(),
                         observedFileNames.size());
```